manticore/platforms/evm.py

```diff
@@ -1750,16 +1750,14 @@ def __setstate__(self, state):
         self._transactions = state['transactions']
         self._initial_block_number = state['initial_block_number']
         self._initial_timestamp = state['_initial_timestamp']
-        self._do_events()
+
+        for _, _, _, _, vm in self._callstack:
```

Comment:
I wonder if you have considered just adding the lacking self._do_events() to close_transaction?

Reply:
Yeah, but that would override/duplicate the registration if the previous vm was already registered, i.e. if the the top-most item was appended in `_open_transaction`. I thought doing it this way would be a bit cleaner and easier to follow.